### ISSUE TITLE
[0.12] Worldgen crash

### ISSUE THREAD

**Issue by Tokeli** [PersonTheCat/CaveGenerator#14]
Forge: 14.23.4.2760

Woopsie!

```
---- Minecraft Crash Report ----
// Hey, that tickles! Hehehe!

Time: 2018-09-28 09:43:24 EDT
Description: Exception ticking world

java.lang.IllegalArgumentException: bound must be positive
    at java.util.Random.nextInt(Random.java:388)
    at com.personthecat.cavegenerator.world.feature.CaveFeatureGenerator.getMatchOnYAxis(CaveFeatureGenerator.java:716)
    at com.personthecat.cavegenerator.world.feature.CaveFeatureGenerator.structureFromDirection(CaveFeatureGenerator.java:346)
    at com.personthecat.cavegenerator.world.feature.CaveFeatureGenerator.generateStructures(CaveFeatureGenerator.java:330)
    at com.personthecat.cavegenerator.world.feature.CaveFeatureGenerator.generate(CaveFeatureGenerator.java:59)
    at net.minecraftforge.fml.common.registry.GameRegistry.generateWorld(GameRegistry.java:167)
    at net.minecraft.world.chunk.Chunk.populate(Chunk.java:1018)
    at net.minecraft.world.chunk.Chunk.populate(Chunk.java:988)
    at net.minecraft.world.gen.ChunkProviderServer.provideChunk(ChunkProviderServer.java:157)
    at net.minecraft.server.management.PlayerChunkMapEntry.providePlayerChunk(PlayerChunkMapEntry.java:126)
    at net.minecraft.server.management.PlayerChunkMap.tick(PlayerChunkMap.java:226)
    at net.minecraft.world.WorldServer.tick(WorldServer.java:227)
    at net.optifine.override.WorldServerOF.tick(WorldServerOF.java:39)
    at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:756)
    at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668)
    at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:252)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
    at java.lang.Thread.run(Thread.java:745)
```

Full crash here: https://paste.dimdev.org/odayufikob.mccrash

**Comment by PersonTheCat** [comment]
Hey, thank you so much for reporting this. This is definitely something I need to fix; however, I bet this can actually be worked around by changing the height settings for one of your structures. They're too low. I can help you figure this out if you share the preset with me in a code block here or using pastebin. 

Sorry about that!

**Comment by Tokeli** [comment]
Oh! I enabled your default `underground_forest` preset, that's all.

I've been flying around **A LOT** as I tweaked the cave sizes over and over though, and I've only had that crash happen once, so it seems to be a pretty rare fluke.

**Comment by PersonTheCat** [comment]
Damn it. I guess I've never seen it because I always treat using the same world and fly in the same direction so I can compare differences. Anyway, I'll fix this soon. Just need to make the number not go negative.

**Comment by P3rf3ctXZer0** [comment]
Yeah I have the same crash - I think.

https://gist.github.com/P3rf3ctXZer0/cea1ace302b0b3985679fb340789eedd

I left the config vanilla.

**Comment by PersonTheCat** [comment]
Damn. Sorry, guys! I know it's been a while, but I've been busy. I'll try to fix this soon, because it should be fairly easy.

**Comment by PersonTheCat** [comment]
Fixed in version 0.13.
